drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -428,31 +418,28 @@ static int _get(gnrc_netdev_t *device, netopt_t opt, void *val, size_t max_len)
             return sizeof(eui64_t);
 
         case NETOPT_PROTO:
-            if (max_len < sizeof(gnrc_nettype_t)) {
-                return -EOVERFLOW;
-            }
+            assert(max_len < sizeof(gnrc_nettype_t));
             *((gnrc_nettype_t *)val) = dev->proto;
             return sizeof(gnrc_nettype_t);
 
         case NETOPT_CHANNEL:
-            if (max_len < sizeof(uint16_t)) {
-                return -EOVERFLOW;
-            }
+            assert(max_len < sizeof(uint16_t));
             ((uint8_t *)val)[1] = 0;
             ((uint8_t *)val)[0] = at86rf2xx_get_chan(dev);
             return sizeof(uint16_t);
 
+        case NETOPT_CHANNEL_PAGE:
+            assert(max_len < sizeof(uint8_t));
```

Comment:
By changing the size this should also be updated in

 `sys/shell/commands/sc_netif.c#228`

    res = gnrc_netapi_get(dev, NETOPT_CHANNEL_PAGE, 0, &u16, sizeof(u16));

    if (res >= 0) {
        printf(" Page: %" PRIu16 " ", u16);
    }

to:

    res = gnrc_netapi_get(dev, NETOPT_CHANNEL_PAGE, 0, &u8, sizeof(u8));

    if (res >= 0) {
        printf(" Page: %" PRIu8 " ", u8);
    }

 `sys/shell/commands/sc_netif.c#574`
`return _netif_set_u16(dev, NETOPT_CHANNEL_PAGE, value);` to `return _netif_set_u8(dev, NETOPT_CHANNEL_PAGE, value);`